### DISCUSSION TITLE
Thoughts about supporting alternate prediction models and more detailed food data

### DISCUSSION THREAD

**Discussion by simonpickering** [Navid200/xDrip#95] [Ideas]
It would be nice if the so called "business logic" prediction model of the app could be swapped for a different model if a user so desires. There has been interest in capturing and using things like food glycaemic index and using that, as well as fat/protein and even caffeine intake. 

It would be good if these models could be easily imported and ideally edited (e.g. by hosting something like a Python runtime internally) rather than needing to install a binary model as an app and communicating with it via Intents.

This would admittedly require some fairly major changes to the input dialog and the database that previously only stored carbohydrate treatments (e.g. deciding on a set of "things" to be stored - protein, fat, carbohydrate, etc., etc.), so by no means trivial.

I would also suggest that XDrip+ should be able to accept meal/drink treatment data from external apps (e.g. a dedicated nutrition tracking app that could provide its own input methods and host a database of meals and food types to avoid this all needing to be performed in XDrip+). This incurs the same costs as detailed above re database and selection of what to store, as well as requiring some changes to allow treatments to be accepted in an Intent.

There is some background info in these issues, which ought to be discussed here rather than there: https://github.com/NightscoutFoundation/xDrip/issues/1525 and https://github.com/NightscoutFoundation/xDrip/issues/1519

**Comment by yoshimo** [comment]
Yea it is an intresting topic.
I currently use the mysugr app to track various things, pictures, ingredients, and additional health data like bloodpressure, pulse, longterm hba1c, weight, sport activity , steps, medizine used, type of the meal, emotional state, special events, keton level. I think xdrip could benefit from more detailed data input.

Dedicated nutrition tracking to steal from ? Maybe Snaq , although i haven't checked its ai powered and partially paid services for quality yet.

**Comment by simonpickering** [comment]
I've started writing a barebones app to provide what I want on the meals and data logging front including free sources of nutrition data, etc. It will likely be some time before it's ready for use tho!

I would be very pleased if we get to a future where the closed source/paid for apps all share data about food intake/exercise/blood glucose/etc. measurements (via e.g. Intents - so just the description of an event, no need to share their USP data sources) and people can choose to pay for different data entry methods/analysis but not necessarily all within a single app - means the apps can focus and makes it more flexible to fit with what people want to get out of it.

I realise this is the standard open source (pipe ;)) dream, but I still have hope it can be made to happen, especially with the move towards people owning their own data and expecting to be able to move it around.

**Reply by yoshimo** [reply]
There is also a German searchengine with an app that can scan Barcodes on products that is specialised in finding carbs for each meal and product. Maybe integrating some international version of wetid would be great
